aws/resource_aws_appconfig_deployment.go

```diff
@@ -0,0 +1,216 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/appconfig"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsAppconfigDeployment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppconfigDeploymentCreate,
+		Read:   resourceAwsAppconfigDeploymentRead,
+		Update: resourceAwsAppconfigDeploymentUpdate,
+		Delete: resourceAwsAppconfigDeploymentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"application_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"environment_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"deployment_strategy_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"configuration_profile_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"configuration_version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 1024),
+				),
+			},
+			"tags": tagsSchema(),
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"deployment_number": {
```

Comment:
I don't see this in the definition in the github issue you linked: 

```
resource "aws_appconfig_deployment" "this" {
  application_id = aws_appconfig_application.this.id
  environment_id = aws_appconfig_environment.this.id
  deployment_strategy_id = aws_appconfig_deployment_strategy.this.id
  configuration_profile_id = aws_appconfig_configuration_profile.this.id
  configuration_version = "1"
  description = "just a sample"
  tags = {
    Name = "AppConfig Deployment"
  }
}
```

Reply:
Ah yeah, we needed it for creating the ARN and stopping the deployment, so it made sense to add it. It's a computed value so it wouldn't show up in the plan the user writes.